.travis.yml

```diff
@@ -0,0 +1,75 @@
+language: php
+
+php:
+  - '7.1'
+  - '7.2'
+  - '7.3'
+
+sudo: required
+
+services:
+  - docker
+  - mysql
+
+mysql:
+  database: sparql_entity_storage_test
+  username: root
+  encoding: utf8
+
+env:
+  global:
+    - SITE_DIR="$HOME/build/testing_site"
+  matrix:
+    - TEST=PHP_CodeSniffer
+    - TEST=8.6.x
+    - TEST=8.7.x
+    - TEST=8.8.x
+
+matrix:
+  fast_finish: true
+  exclude:
+    - php: '7.2'
+      env: TEST=PHP_CodeSniffer
+    - php: '7.3'
+      env: TEST=PHP_CodeSniffer
+
+before_install:
+  - composer --verbose self-update
+  - mkdir -p ${SITE_DIR}/web/modules
+  - cd ${SITE_DIR}
+
+  ### PHP_CodeSniffer only.
+
+  # We only need the PHP Code Sniffer (and the Drupal coder).
+  - test ${TEST} == "PHP_CodeSniffer" && composer require drupal/coder || true
+
+  ### PHPUnit only (till the end).
+
+  # Deploy the codebase.
+  - test ${TEST} == "PHP_CodeSniffer" || cp ${TRAVIS_BUILD_DIR}/tests/travis-ci/fixtures/composer.json.dist ${SITE_DIR}/composer.json
+  - test ${TEST} == "PHP_CodeSniffer" || cp ${TRAVIS_BUILD_DIR}/tests/travis-ci/fixtures/phpunit.xml.dist ${SITE_DIR}/web/phpunit.xml
+  - test ${TEST} == "PHP_CodeSniffer" || perl -i -pe's/\$\{([^}]+)\}/$ENV{$1}/' composer.json
+  - test ${TEST} == "PHP_CodeSniffer" || composer require drupal/core:${TEST} --no-interaction
+  # Virtuoso setup.
+  - test ${TEST} == "PHP_CodeSniffer" || mkdir ${SITE_DIR}/virtuoso
+  - test ${TEST} == "PHP_CodeSniffer" || docker run --name virtuoso -p 8890:8890 -p 1111:1111 -e SPARQL_UPDATE=true -v ${SITE_DIR}/virtuoso:/data -d tenforce/virtuoso
+  # Create the MySQL database.
+  - test ${TEST} == "PHP_CodeSniffer" || mysql -e 'CREATE DATABASE sparql_entity_storage_test'
+  # Install Drupal.
+  - test ${TEST} == "PHP_CodeSniffer" || ./vendor/bin/drush site:install testing --yes --root=${SITE_DIR}/web --db-url=mysql://root:@127.0.0.1/sparql_entity_storage_test
+  # Add the SPARQL connection to settings.php.
+  - test ${TEST} == "PHP_CodeSniffer" || chmod 0775 ${SITE_DIR}/web/sites/default/settings.php
+  - test ${TEST} == "PHP_CodeSniffer" || cat ${TRAVIS_BUILD_DIR}/tests/travis-ci/fixtures/connection.txt >> ${SITE_DIR}/web/sites/default/settings.php
+  # Enable the 'sparql_entity_storage' module.
+  - test ${TEST} == "PHP_CodeSniffer" || ./vendor/bin/drush pm:enable sparql_entity_storage --yes --root=${SITE_DIR}/web
+  # Start the webserver for browser tests.
+  - cd ${SITE_DIR}/web
+  - test ${TEST} == "PHP_CodeSniffer" || nohup php -S localhost:8888 > /dev/null 2>&1 &
+  # Wait until the web server is responding.
+  - test ${TEST} == "PHP_CodeSniffer" || until curl -s localhost:8888; do true; done > /dev/null
+  - cd ..
```

Comment:
Instead of messing with `test ... || ...`, why not moving the whole logic in the script. See 
 https://github.com/ec-europa/rdf_entity/blob/8.x-1.x/.travis.yml and https://github.com/ec-europa/rdf_entity/blob/8.x-1.x/tests/travis-ci/scripts/run_tests.sh.

Also, as in https://github.com/ec-europa/rdf_entity/blob/8.x-1.x/.travis.yml, I would disable xdebug in tests. On rare cases, when we need debugging, we'll enabled it.